### PR TITLE
Enable modular build for clang 3.9.

### PR DESCRIPTION
**Pull request by y4vor**
b/265873566

Change-Id: I788d62031e6550a7bad43810537e8f036c44f1f2

### CODE REVIEW

File: starboard/linux/x64x11/clang/3.9/cobalt/configuration.py [hunk 1]
```diff
@@ -29,6 +29,13 @@ def GetTestFilters(self):
 
   # A map of failing or crashing tests per target.
   __FILTERED_TESTS = {  # pylint: disable=invalid-name
+      'blackbox': [
+        'cancel_sync_loads_when_suspended',
```
Comment (andrewsavage1) [inline starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:33]: I assume we should have a TODO to get the rest of these working? Also, we can probably have the same filters for linux-x64x11 modular bb tests and they would start working?

File: starboard/linux/x64x11/clang/3.9/platform_configuration/BUILD.gn [hunk 2]
```diff
@@ -180,10 +183,15 @@ config("no_pedantic_warnings") {
...
 config("platform_configuration") {
-  configs = [
-    "//starboard/linux/shared/platform_configuration",
-    "//starboard/linux/x64x11/shared/platform_configuration:libraries",
-    "//starboard/build/config/sabi",
-    ":compiler_flags",
-  ]
+  if (current_toolchain == default_toolchain &&
+      build_with_separate_cobalt_toolchain) {
+    configs = [ "//starboard/evergreen/x64/platform_configuration" ]
```
Comment (andrewsavage1) [inline starboard/linux/x64x11/clang/3.9/platform_configuration/BUILD.gn:188]: This should point to the modular config instead

File: starboard/nplb/BUILD.gn [hunk 3]
```diff
@@ -265,8 +265,8 @@ target(gtest_target_type, "nplb") {
     cflags = [ "-Wno-enum-constexpr-conversion" ]
   }
 
-  #  TODO: b/297808555 - Add these tests for windows based platform modular builds.
-  if (sb_is_modular && !sb_is_evergreen && is_host_win) {
+  #  TODO: b/297808555 - Add these tests for modular builds.
+  if (sb_is_modular && !sb_is_evergreen) {
     sources -= [
       "maximum_player_configuration_explorer.cc",
```
Comment (andrewsavage1) [inline starboard/nplb/BUILD.gn:271]: This also removes the tests from current modular linux builds which I would prefer not to do.

Comment (y4vor) [reply starboard/nplb/BUILD.gn:271]: Yeah, I know.
Will add a variable that platforms can turn on to escape those test.  I was hesitant to add a new variable just for that, but I think more test coverage would be better.

File: starboard/linux/x64x11/clang/3.9/cobalt/configuration.py (hunk 1 above)
Comment (y4vor) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:33]: Added a tracking bug.

Comment (andrewsavage1) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:34]: As part of this I want to move these filters to a shared location—I'm trying this out in https://github.com/youtube/cobalt/pull/1398 right now, if that works we can move this

Comment (y4vor) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:34]: Don't we run the test on Linux Modular today? 
I think we should disable only for clang 3.9 modular if we can.

Comment (andrewsavage1) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:34]: We don't run black box tests at all for modular linux today, see the linked PR
It looks to me like with these filters in place, modular linux-x64x11 fails just on `test_wasm_basic`, does that pass for this platform or is it not being run?

Comment (y4vor) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:34]: The `test_wasm_basic` passes on clang 3.9 modular

https://github.com/youtube/cobalt/actions/runs/6396825020/job/17363792926

========================================


[1654:10-03 18:30.730...:INFO:black_box_tests.py(143)] Running WasmBasicTest
[1654:10-03 18:30.731...:WARNING:wasm_basic_test.py(42)] Blackbox tests disabled for platform:linux-x64x11-clang-3-9
[1654:10-03 18:30.731...:INFO:black_box_tests.py(148)] Done WasmBasicTest
[1654:10-03 18:30.731...:INFO:black_box_tests.py(142)] 


test_wasm_basic (cobalt.black_box_tests.tests.wasm_basic_test.WasmBasicTest) ... o

Comment (andrewsavage1) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py:34]: Got it, let's add all these filters to the shared linux filters and I can add the filter for the wasm test on top of that just for the linux-x64x11 builds, hopefully we can look into these failures soon

File: starboard/linux/x64x11/clang/3.9/platform_configuration/configuration.gni [hunk 4]
```diff
@@ -12,38 +12,48 @@
...
-import("//starboard/build/config/base_configuration.gni")
+if (current_toolchain == default_toolchain &&
+    build_with_separate_cobalt_toolchain) {
+  import("//starboard/evergreen/x64/platform_configuration/configuration.gni")
+  platform_tests_path =
+      "//starboard/linux/shared:starboard_platform_tests($starboard_toolchain)"
+  cobalt_font_package = "standard"
 
-nasm_exists = true
+  sb_has_unused_symbol_issue = true
```
Comment (andrewsavage1) [inline starboard/linux/x64x11/clang/3.9/platform_configuration/configuration.gni:22]: Is this something we want to set only for the cobalt toolchain? My only thought is that someone might try to use this in the starboard toolchain and be surprised with how it's set, but maybe we don't want to use it for targets being built by the starboard toolchain at all?

Also remember to make the playstation change internally first

File: starboard/linux/x64x11/clang/3.9/cobalt/configuration.py (hunk 1 above)
Comment (y4vor) [reply starboard/linux/x64x11/clang/3.9/cobalt/configuration.py]: Ok, moved them in the shared linux modular filter.

File: starboard/linux/x64x11/clang/3.9/platform_configuration/configuration.gni (hunk 4 above)
Comment (y4vor) [reply starboard/linux/x64x11/clang/3.9/platform_configuration/configuration.gni:22]: The issue showed up only with the cobalt toolchain. Could be related to the stripping unused symbols when building a shared library. I would prefer to keep the scope narrow and match the observed issues.

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**